Add non-IID-ness metrics for label skew

damjimenezgu opened the pull request:
<!--
Thank you for opening a pull request (PR)!

Contribution guidelines: https://github.com/adap/flower/blob/main/CONTRIBUTING.md
-->

## Issue

### Description

<!--
Describe the problem addressed by this PR.

This is the first commit to add the metrics (Hellinger and Jensen-Shannon distances) to measure non-IID-ness for labels skew for classification and regression tasks.
-->

### Related issues/PRs

<!--
Link issues and/or PRs that are related to this PR.


-->

## Proposal

### Explanation

<!--
Explain the changes and how they improve the issue described above.

The folder "metrics" is included inside the path "datasets/flwr_datasets" containing 5 ".py" files.
-->

### Checklist

- [ ] Implement proposed change
- [ ] Write tests
- [ ] Update [documentation](https://flower.dev/docs/writing-documentation.html)
- [ ] Update the changelog entry below
- [ ] Make CI checks pass
- [ ] Ping maintainers on [Slack](https://flower.dev/join-slack/) (channel `#contributions`)

<!--
Inside the following 'Changelog entry' section, you should put the description of your changes that will be added to the changelog alongside your PR title.

If the section is completely empty (without any token) or non-existant, the changelog will just contain the title of the PR for the changelog entry, without any description. If the section contains some text other than tokens, it will use it to add a description to the change. If the section contains one of the following tokens it will ignore any other text and put the PR under the corresponding section of the changelog:

<general> is for classifying a PR as a general improvement.
<skip> is to not add the PR to the changelog
<baselines> is to add a general baselines change to the PR
<examples> is to add a general examples change to the PR
<sdk> is to add a general sdk change to the PR
<simulations> is to add a general simulations change to the PR

Note that only one token should be used.
-->

### Changelog entry



### Any other comments?

<!--
Please be aware that it may take some time until the maintainers can review the PR.
Smaller PRs with good descriptions can be considered much more easily.

If you have an urgent request or question, please use the Flower Slack:

    https://flower.dev/join-slack/ (channel: #contributions)

Thank you for contributing to Flower!
-->

adam-narozniak reviewed (changes requested):
Hi, thanks for the contribution. Let me start with a high-level comments.

1. What is the reason you decided to make these metrics classes? They don't hold any state or are not cumulative in any sense along time/process.
2. We can split the metric by the task since there's at least `num_bins` as an additional parameter.
3. We need to check if the target values match the task.
4. Let's prioritize clarity over short names.
5. Some docs are missing. (I'll approve the CI test running but can also see it locally by running dev/format.sh and dev/test.sh)
6. The same goes for hint types.

adam-narozniak commented on datasets/flwr_datasets/metrics/__init__.py:
```diff
@@ -0,0 +1,21 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Flower Datasets Partitioner package."""
```
Incorrect

damjimenezgu replied:
Yes, I corrected it to "Flower Datasets Metrics package"

adam-narozniak commented on datasets/flwr_datasets/metrics/get_distros.py:
```diff
@@ -0,0 +1,108 @@
+import numpy as np
+
```
no docs

damjimenezgu replied:
Yes, I solved it and now there are no missing docs

damjimenezgu commented:
Hi,  Adam. Thanks a bunch for your feedback. Let me answer to each one of your comments (in _italic_)

1. What is the reason you decided to make these metrics classes? They don't hold any state or are not cumulative in any sense along time/process. _ANS: I wanted to keep the same format of `partitioner` and `resplitter`, but you are right. I have changed everything to only functions (instead of classes)._
2. We can split the metric by the task since there are at least num_bins as an additional parameter. _ANS: Yes, I split each metric by task as suggested_
3. We should also have a high-level function that takes the Partitioner object, which simplifies the user experience. _ANS: Yes, I have added it, and now it is similar to the way how the classes from `partitioner` and `resplitter` are loaded_
4. We need to check if the target values match the task. _ANS: Yes, I included the validation (raise method) in each metric function_
5. Let's prioritize clarity over short names. _ANS: Yes, I modified all the functions by keeping the names clear instead of keeping them short_
6. Some docs are missing. (I'll approve the CI test running but can also see it locally by running dev/format.sh and dev/test.sh): _ANS: Yes, now there are no missing docs, as checked with dev/format.sh and dev/test.sh_
7. The same goes for hint types. _ANS: Yes, I included the hint types for all the functions_
8. We need simpler tests. _ANS: It is unclear to me what you mean by "simpler tests". Could you please elaborate and provide some examples?_
9. We need to test edge cases. _ANS: Yes, I included two edge cases for each metric in the unitary tests (minimum and maximum value for the metrics)_
